cblox_ros/src/submap_server.cc

```diff
@@ -0,0 +1,89 @@
+#include "cblox_ros/submap_server.h"
+
+namespace cblox {
+
+template <>
+void SubmapServer<TsdfSubmap>::visualizeSlice(const SubmapID submap_id) const {
+  if (sdf_slice_pub_.getNumSubscribers() < 1) {
+    return;
+  }
+  if (!submap_collection_ptr_->exists(submap_id)) {
+    ROS_WARN("[CbloxServer] Submap %d does not exist.", submap_id);
+    return;
+  }
+
+  if (verbose_) {
+    ROS_INFO("[CbloxServer] Visualizing TSDF slice of submap %d at height %.2f",
+             submap_id, slice_height_);
+  }
+
+  visualization_msgs::MarkerArray marker_array;
+  visualization_msgs::Marker vertex_marker;
+  vertex_marker.header.frame_id = world_frame_;
+  vertex_marker.ns = std::string("esdf_slice_") + std::to_string(submap_id);
+  vertex_marker.ns = "esdf_slice";
+  vertex_marker.type = visualization_msgs::Marker::CUBE_LIST;
+  Transformation pose =
+      submap_collection_ptr_->getSubmapPtr(submap_id)->getPose();
+  vertex_marker.pose.orientation.w = 1.0;
+  vertex_marker.scale.x =
+      submap_collection_ptr_->getActiveTsdfMapPtr()->voxel_size();
+  vertex_marker.scale.y = vertex_marker.scale.x;
+  vertex_marker.scale.z = vertex_marker.scale.x;
+  geometry_msgs::Point point_msg;
+  std_msgs::ColorRGBA color_msg;
+  color_msg.r = 0.0;
+  color_msg.g = 0.0;
+  color_msg.b = 0.0;
+  color_msg.a = 1.0;
+
+  float max_dist = 0.6;
+  nh_private_.param("truncation_distance", max_dist, max_dist);
```

Comment:
Like this, the ROS master gets queried every time `SubmapServer<TsdfSubmap>::visualizeSlice(...)` is executed.
Do you intend to change this param at runtime? If yes, it might be worth using [cached params](http://wiki.ros.org/roscpp/Overview/Parameter%20Server#Cached_Parameters). Otherwise it might be nicer to make `max_dist` a class variable and move its loading to [getParametersFromRos()](https://github.com/ethz-asl/cblox/blob/85c44b4351a91745801c7bd4c15245d3e8103b96/cblox_ros/include/cblox_ros/submap_server.h#L110).

Reply:
Done!